src/main/java/EventsAndCommands/CommandLoader.java

```diff
@@ -0,0 +1,222 @@
+package EventsAndCommands;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.jagrosh.jdautilities.command.Command;
+
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+
+import org.reflections.*;
+import com.mongodb.client.MongoDatabase;
+import com.jagrosh.jdautilities.commons.waiter.EventWaiter;
+
+/**
+ * CommandHandler
+ */
+public class CommandLoader {
+
+    private List<Command> loadedCommands;
+    private List<ListenerAdapter> loadedListeners;
+    private MongoDatabase db;
+    private Properties config;
+    private EventWaiter waiter;
+
+    public CommandLoader(MongoDatabase db, Properties config, EventWaiter waiter) {
+        loadedCommands = new ArrayList<>();
+        loadedListeners = new ArrayList<>();
+        setDb(db);
+        setConfig(config);
+        setWaiter(waiter);
+    }
+
+    public void loadCommands() {
+        Reflections reflections = new Reflections(this.getClass().getPackageName());
+
+        Set<Class<? extends Command>> cmds = reflections.getSubTypesOf(Command.class);
+        cmds = cmds.stream()
+            .filter(x -> x.getAnnotation(IgnoreCommand.class) == null)
+            .collect(Collectors.toSet());
+
+        List<Command> instances = new ArrayList<>();
+
+        for (var cmd : cmds) {
+            List<Constructor<?>> ctors = Arrays.asList(cmd.getConstructors());
+            for (var ctor : ctors) {
+                if (hasAcceptableParams(ctor)) {
+                    try {
+                        instances.add(injectCommand(ctor));
+                    } catch (InvocationTargetException e){
+                        e.getCause().printStackTrace();
```

Comment:
Why do you only print the cause? It probably has a good reason, but what is it?

Reply:
InvocationTargetException wraps another Exception that was thrown inside the code that got invoked via reflection. In this case getCause() returns the actually thrown exception and not the wrapper.

See: <https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/lang/reflect/InvocationTargetException.html#getCause()>

Reply:
I am aware, but then you lose the context where exactly it was invoked from:
```bash
$ javac Test.java && java Test
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at Test.main(Test.java:9)
Caused by: java.lang.RuntimeException: Hey
	at Test.test(Test.java:4)
	... 5 more

# Changed to getCause().printStackTrace()

$ javac Test.java && java Test
java.lang.RuntimeException: Hey
	at Test.test(Test.java:4)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at Test.main(Test.java:9)

```

Reply:
Fixed it, didn'T notice that it still prints the wrapped cause since my IDE wraps stacktraces